src/include/txn_inline.h

```diff
@@ -363,34 +363,54 @@ __wt_txn_op_delete_apply_prepare_state(WT_SESSION_IMPL *session, WT_REF *ref, bo
  * __txn_op_delete_commit_apply_page_del_timestamp --
  *     Apply the correct start and durable timestamps to the page delete structure.
  */
-static WT_INLINE void
-__txn_op_delete_commit_apply_page_del_timestamp(WT_SESSION_IMPL *session, WT_REF *ref)
+static WT_INLINE int
+__txn_op_delete_commit_apply_page_del_timestamp(WT_SESSION_IMPL *session, WT_TXN_OP *op)
 {
+    WT_ADDR_COPY addr;
+    WT_DECL_RET;
     WT_PAGE_DELETED *page_del;
+    WT_REF *ref;
     WT_TXN *txn;
+    bool addr_found;
 
+    ref = op->u.ref;
     txn = session->txn;
     page_del = ref->page_del;
 
     WT_ASSERT(session, WT_REF_GET_STATE(ref) == WT_REF_LOCKED);
 
     if (page_del != NULL && page_del->timestamp == WT_TS_NONE) {
+
+        /* Validate the commit timestamp against the maximum durable timestamp on the page. */
+        WT_ENTER_GENERATION(session, WT_GEN_SPLIT);
+        WT_WITH_BTREE(session, op->btree, addr_found = __wt_ref_addr_copy(session, ref, &addr));
+        if (addr_found)
+            ret = __wt_txn_timestamp_usage_check(session, op, txn->commit_timestamp,
+              WT_MAX(addr.ta.newest_start_durable_ts, addr.ta.newest_stop_durable_ts));
+        WT_LEAVE_GENERATION(session, WT_GEN_SPLIT);
```

Comment:
Shouldn't we leave the generation sooner (before checking the timestamps)?

Reply:
As soon as we leave the generation, the `addr.addr` can be freed. But anyway we are not using that memory here, so the generation can be released sooner when we get the copy, but not releasing it sooner also doesn't cause any problems other than holding the memory for some more time if it is freed in parallel.

Reply:
great makes sense to leave it this way, thanks!